recipes/prometheus-cpp/all/conanfile.py

```diff
@@ -0,0 +1,120 @@
+        self._cmake = CMake(self)
+        self._cmake.definitions["USE_THIRDPARTY_LIBRARIES"] = False
+        self._cmake.definitions["ENABLE_TESTING"] = False
+        self._cmake.definitions["OVERRIDE_CXX_STANDARD_FLAGS"] = False
+
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        self._cmake.definitions["ENABLE_PULL"] = self.options.with_pull
+        self._cmake.definitions["ENABLE_PUSH"] = self.options.with_push
+        self._cmake.definitions["ENABLE_COMPRESSION"] = self.options.with_compression
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(os.path.join(self._source_subfolder, "LICENSE"), dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "prometheus-cpp"
+        self.cpp_info.names["cmake_find_package_multi"] = "prometheus-cpp"
+
+        self.cpp_info.components["prometheus-cpp-core"].names["cmake_find_package"] = "core"
+        self.cpp_info.components["prometheus-cpp-core"].names["cmake_find_package_multi"] = "core"
+        self.cpp_info.components["prometheus-cpp-core"].libs = ["prometheus-cpp-core"]
+        if self.settings.os == "Linux":
+            self.cpp_info.components["prometheus-cpp-core"].system_libs = ["pthread", "rt"]
+
+        if self.options.with_push:
+            self.cpp_info.components["prometheus-cpp-push"].names["cmake_find_package"] = "push"
+            self.cpp_info.components["prometheus-cpp-push"].names["cmake_find_package_multi"] = "push"
+            self.cpp_info.components["prometheus-cpp-push"].libs = ["prometheus-cpp-push"]
+            self.cpp_info.components["prometheus-cpp-push"].requires = [
+                "prometheus-cpp-core",
+                "libcurl::libcurl",
+            ]
+            if self.settings.os == "Linux":
+                self.cpp_info.components["prometheus-cpp-push"].system_libs = ["pthread", "rt"]
+
+        if self.options.with_pull:
+            self.cpp_info.components["prometheus-cpp-pull"].names["cmake_find_package"] = "pull"
+            self.cpp_info.components["prometheus-cpp-pull"].names["cmake_find_package_multi"] = "pull"
+            self.cpp_info.components["prometheus-cpp-pull"].libs = ["prometheus-cpp-pull"]
+            self.cpp_info.components["prometheus-cpp-pull"].requires = [
+                "prometheus-cpp-core",
+                "civetweb::civetweb-cpp"
+            ]
+            if self.options.with_compression:
+                self.cpp_info.components["prometheus-cpp-pull"].requires.append("zlib::zlib")
+            if self.settings.os == "Linux":
+                self.cpp_info.components["prometheus-cpp-pull"].system_libs = ["pthread", "rt"]
```

Comment:
is this needed? they are specified with the core already

Reply:
explicit is better than implicit. They are direct dependencies of these components, all these dependencies are private, so if conan can implement public/private property in the future, this recipe clearly shows dependencies.